consensus/core/src/authority_node.rs

```diff
@@ -93,40 +108,79 @@ impl AuthorityNode {
     }
 
     #[allow(unused)]
-    pub fn transactions_client(&self) -> Arc<TransactionsClient> {
+    pub fn transactions_client(&self) -> Arc<TransactionClient> {
         self.transactions_client.clone()
     }
 }
 
+/// Authority's network interface.
+struct AuthorityService {
+    _context: Arc<Context>,
+    block_verifier: Arc<dyn BlockVerifier>,
+}
+
+#[async_trait]
+impl NetworkService for AuthorityService {
+    async fn handle_send_block(
+        &self,
+        peer: AuthorityIndex,
+        serialized_block: Bytes,
+    ) -> ConsensusResult<()> {
+        // TODO: dedup block verifications, with fetched blocks.
+        let signed_block: SignedBlock = bcs::from_bytes(&serialized_block)?;
+        if peer != signed_block.author() {
+            return Err(ConsensusError::UnexpectedAuthority(
+                signed_block.author(),
+                peer,
+            ));
+        }
+        self.block_verifier.verify_metadata(&signed_block)?;
```

Comment:
Is there a time when we wouldnt want to verify both the metadata and the transactions? Maybe we just have block_verifier.verify() that internally calls both of these methods?